llvm/lib/Transforms/Utils/BreakCriticalEdges.cpp

```diff
@@ -175,6 +175,8 @@ llvm::SplitKnownCriticalEdge(Instruction *TI, unsigned SuccNum,
   // Create our unconditional branch.
   BranchInst *NewBI = BranchInst::Create(DestBB, NewBB);
   NewBI->setDebugLoc(TI->getDebugLoc());
+  if (auto *LoopMD = TI->getMetadata(LLVMContext::MD_loop))
```

Comment:
Ideally it should go into a separate PR, but the issue of not copying llvm.loop is being exposed on during a call to reg2mem pass, which (if I understood correctly) is in semi-deprecated state and it's not trivial to test it.